### ISSUE TITLE
Opaque definition and purpose of `proto/varlogpb.(ReplicaDescriptor).Path`

### ISSUE THREAD

**Issue by ijsong** [kakao/varlog#124]
TL;DR
We need to rename `proto/varlogpb.(ReplicaDescriptor).Path` to `proto/varlogpb.(ReplicaDescriptor).StorageNodePath` to clarify the definition and purpose. It is also necessary to add a new field `proto/varlogpb.(ReplicaDescriptor).DataPath` to indicate the directory where the log stream replica stores its data.

It looks `proto/varlogpb.(ReplicaDescriptor).Path` pretty obscure for me. Since the type `proto/varlogpb.(ReplicaDescriptor)` represents where a log stream replica is, the field `Path` shall be a directory where the log stream replica stores its data. In some use cases, it doesn't.

For example, the type `proto/varlogpb.(LogStreamDescriptor)` which represents the log stream metadata registered to the metadata repository contains the field `Replicas` of type `proto/varlogpb.(ReplicaDescriptor)`. As opposed to the field's name, it indicates the directory where the storage node stores sub-directories for replicas. For more concrete example, it is `/data/cid_1_snid_1` rather than `/data/cid_1_snid_1/tpid_1_lsid_1`. However, if storage node metadata is fetched from a storage node through RPC `varlog.snpb.Management/GetMetadata`, `proto/snpb.(LogStreamReplicaMetadataDescriptor).Path` is `/data/cid_1_snid_1/tpid_1_lsid_1`. Although they are different fields, users can be confused. Moreover, by #112, the admin server can return both directories for RPC `varlog.vmspb.ClusterManager/ListStorageNodesRequest` according to the status of admin server. If the memory of the admin server has no information about the storage node it uses the metadata from the metadata repository, otherwise, it uses the metadata from the storage node.
